staging/src/k8s.io/controller-manager/config/v1beta1/types.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LeaderMigrationConfiguration provides versioned configuration for all migrating leader locks.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type LeaderMigrationConfiguration struct {
```

Comment:
Does any consumers of `LeaderMigrationConfiguration` need to be updated to use the new v1beta1 API now?

Reply:
Or I guess not because we only deal with internal types inside our components?

Reply:
The original `v1alpha` version is kept as of this time. I will update the doc on k/website after this PR is merged.

Reply:
Short answer: no. Because for now the only usage of this group would be parsing a yaml file, the conversion is taken care for free.

Reply:
GenericControllerManagerConfiguration was the only other type I could find using it (See above). Its one of the reasons I was curious about it getting updated.